low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -0,0 +1,322 @@
+
+    def get_cluster_running_nodes(self, msg_broker : str):
+        if(msg_broker == 'rabbitmq'):
+            return self.get_rabbitmq_cluster_nodes()
+        else:
+            raise SetupError(errno.EINVAL,
+                        "Provided message broker '%s' is not supported",
+                        msg_broker)
+
+    def process(self):
+        cmd = "config"
+        self.role = Conf.get('global_config', 'release>setup')
+        if(cmd == "config"):
+            self.config_sspl()
+        else:
+            raise SetupError(errno.EINVAL, "cmd val should be config")
+
+        # Get the version. Output can be 3.3.5 or 3.8.9 or in this format
+        rmq_cmd = "rpm -qi rabbitmq-server"
+        output, error, returncode = SimpleProcess(rmq_cmd).run()
+        if returncode:
+            raise SetupError(returncode, error)
+        self.rabbitmq_version = re.search(  r'Version     :\s*([\d.]+)',
+                                            str(output)).group(1)
+
+        # Get the Major release version parsed. (Eg: 3 from 3.8.9)
+        self.rabbitmq_major_release = self.rabbitmq_version[0]
+
+        # Get the Minor release version parsed. (Eg: 3.8 from 3.8.9)
+        self.rabbitmq_minor_release = self.rabbitmq_version[:3]
+
+        # Get the Maitenance release version parsed from minor release.
+        # (Eg: 8 from 3.8)
+        self.rabbitmq_maintenance_release = self.rabbitmq_minor_release[-1]
+
+        # Skip this step if sspl is being configured for node replacement 
+        # scenario as consul data is already
+        # available on healthy node
+        # Updating RabbitMQ cluster nodes.
+        # In node replacement scenario, avoiding feeding again to avoid
+        # over writing already configured values
+        # with which rabbitmq cluster may have been created
+        if not consts.REPLACEMENT_NODE_ENV_VAR_FILE:
+            message_broker="rabbitmq"
+            # Get the running nodes from a cluster
+            pout = self.get_cluster_running_nodes(message_broker)
+
+            # Update cluster_nodes key in consul
+            if consts.PRODUCT_NAME == 'LDR_R1':
+                host = os.getenv('CONSUL_HOST', consts.CONSUL_HOST)
+                port = os.getenv('CONSUL_PORT', consts.CONSUL_PORT)
+                try:
+                    consul_conn = consul.Consul(host=host, port=port)
+                    consul_conn.kv.put(
+                            "sspl/config/RABBITMQCLUSTER/cluster_nodes", pout)
+                except Exception:
+                    raise
+            else:
+                Conf.set('sspl', 'RABBITMQCLUSTER>cluster_nodes', pout)
+                Conf.save('sspl')
+            
+        # Skip this step if sspl is being configured for node replacement
+        # scenario as consul data is already
+        # available on healthy node
+        # Updating build requested log level
+        if not consts.REPLACEMENT_NODE_ENV_VAR_FILE:
+            log_level = None
```

Comment:
This variable assignment is not necessary.
If you would like to keep still, set this as default 'INFO'.